yarn-project/aztec/src/cli/chain_l2_config.ts

```diff
@@ -0,0 +1,60 @@
+export type NetworkNames = 'testnet-ignition';
+
+export type L2ChainConfig = {
+  l1ChainId: number;
+  ethereumSlotDuration: number;
+  aztecSlotDuration: number;
+  aztecEpochDuration: number;
+  aztecProofSubmissionWindow: number;
+  testAccounts: boolean;
+  p2pEnabled: boolean;
+  p2pBootstrapNodes: string[];
+  registryAddress: string;
+};
+
+export const testnetIgnitionL2ChainConfig: L2ChainConfig = {
+  l1ChainId: 11155111,
+  ethereumSlotDuration: 12,
+  aztecSlotDuration: 36,
+  aztecEpochDuration: 32,
+  aztecProofSubmissionWindow: 64,
+  testAccounts: true,
+  p2pEnabled: true,
+  p2pBootstrapNodes: [],
+  registryAddress: '0x12b3ebc176a1646b911391eab3760764f2e05fe3',
+};
+
+export async function getBootnodes(networkName: NetworkNames) {
+  const url = `http://static.aztec.network/${networkName}/bootnodes.json`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Failed to fetch basic contract addresses from ${url}`);
+  }
+  const json = await response.json();
+  return json['bootnodes'];
+}
+
+export async function getL2ChainConfig(networkName: NetworkNames): Promise<L2ChainConfig | undefined> {
+  if (networkName === 'testnet-ignition') {
+    const config = { ...testnetIgnitionL2ChainConfig };
+    config.p2pBootstrapNodes = await getBootnodes(networkName);
+    return config;
+  }
+  return undefined;
+}
+
+export async function enrichEnvironmentWithChainConfig(networkName: NetworkNames) {
+  const config = await getL2ChainConfig(networkName);
+  if (!config) {
+    return;
+  }
+  process.env['ETHEREUM_SLOT_DURATION'] = config.ethereumSlotDuration.toString();
+  process.env['AZTEC_SLOT_DURATION'] = config.aztecSlotDuration.toString();
+  process.env['AZTEC_EPOCH_DURATION'] = config.aztecEpochDuration.toString();
+  process.env['AZTEC_PROOF_SUBMISSION_WINDOW'] = config.aztecProofSubmissionWindow.toString();
+  process.env['P2P_BOOTSTRAP_NODES'] = config.p2pBootstrapNodes.join(',');
+  process.env['TEST_ACCOUNTS'] = config.testAccounts.toString();
+  process.env['P2P_ENABLED'] = config.p2pEnabled.toString();
+  process.env['L1_CHAIN_ID'] = config.l1ChainId.toString();
+  process.env['REGISTRY_CONTRACT_ADDRESS'] = config.registryAddress;
```

Comment:
Feels like we should be changing the config object, rather than process.env directly, but given how chaotic our config is now this seems like the safest option.

That said, wdyt about only setting these variables if they were not explicitly set by the user (ie they're empty)?

Reply:
Not sure, the point of these is that they kind of have to be fixed. In fact I'd like to reduce it to about 3 and retrieve the rest from L1 etc.

Reply:
Second thoughts, will do.

Reply:
Done